.github/workflows/build-jekyll.yml

```diff
@@ -0,0 +1,34 @@
+name: Build and Deploy to GitHub Pages
+
+on:
+  push:
+    branches:
+      - master
```

Comment:
Can we verify that this workflow runs smoothly on our repo?

Perhaps you can push an equivalent change directly to a non-master branch first, and see what happens to the `gh-pages` branch.

Reply:
I thought that the `GITHUB_TOKEN` secret was being blocked due to this PR being from a fork. Seems to [also happen ](https://github.com/AY2021S1-CS2103T-W16-3/tp/runs/1159335401)with branches on this repo. :/

Reply:
Apparently, `GITHUB_TOKEN` cannot be used; must be a GitHub user's personal access token.

Reply:
CI job is able to run now. Tested on a branch of this repo as PRs from forks are unable to access secrets.

See https://github.com/AY2021S1-CS2103T-W16-3/tp/actions/runs/270453191.

Reply:
Ok, as seen from https://github.com/AY2021S1-CS2103T-W16-3/tp/commit/65e48f7a1f38c68e643cb29f8d29e0951f06bc4c the CI is able to build and deploy GitHub Pages properly.